editor.planx.uk/src/@planx/components/SetValue/Editor.tsx

```diff
@@ -6,13 +7,41 @@ import ModalSection from "ui/editor/ModalSection";
 import ModalSectionContent from "ui/editor/ModalSectionContent";
 import Input from "ui/shared/Input";
 import InputRow from "ui/shared/InputRow";
+import Radio from "ui/shared/Radio";
 
 import { parseSetValue, SetValue } from "./model";
 
 type Props = EditorProps<TYPES.SetValue, SetValue>;
 
 export default SetValueComponent;
 
+const DescriptionText: React.FC<SetValue> = ({ fn, val, operation }) => {
+  if (!fn || !val) return null;
+
+  switch (operation) {
+    case "replace":
+      return (
+        <Typography mb={2}>
+          Any existing value for <strong>{fn}</strong> will be replaced by{" "}
+          <strong>{val}</strong>
+        </Typography>
+      );
+    case "append":
+      return (
+        <Typography mb={2}>
+          Any existing value for <strong>{fn}</strong> will have{" "}
+          <strong>{val}</strong> appended to it
+        </Typography>
+      );
+    case "remove":
+      return (
+        <Typography mb={2}>
+          Any existing value for <strong>{fn}</strong> will be removed
```

Comment:
Can `remove` also support an `fn` _and_ `val`  ?

Eg see this example https://opensystemslab.slack.com/archives/C5Q59R3HB/p1701766927022339 - ideally this solution let's us remove need for project types like `not.alter.roofLight` etc !

Reply:
Change now made 👍

Reply:
Revisiting this one and realising the `val` suggestion only makes sense when the `fn` is an array of schema-controlled `val` - we'll need to keep thinking about this for eg `NumberInputs` which set can any string. In that case, I think expected behavior would be to remove based on `fn` alone!

Similar for "replace" - currently it seems that string passport `fns` are converted to an array if a `SetValue` attempts to "Replace" them!

Reply:
This has now been addressed 👍

The `SetValue` will always convert the output of a _modified_ value to an array - this means that it's outputs can be used for automations. If an operation cannot be completed, or is nonsensical (e.g. removing an item from a single string value) the output will not be modified.